### PR TITLE
Sanitize Function value objects

### PR DESCRIPTION
**Pull request by drkameleon**
# Description

Following @beef331's line, let's deal with another elephant in the room, whose revamp is long due: the **Function** values.

Size-wise, each and every one of them (and no, they are not by any definition "few": only the standard library ones are currently 320!) weighs 168 bytes ❗ .

### The problem:
- [x] There are different parts of the object that... are good to be there, but we mostly don't need them (largely, documentation-related); nevertheless, they still add a lot of bulk to our Function value objects. We could wrap all of them in another ref object. Namely: `.args`, `.attrs`, `.returns` & `.example`.
- [x] ~~There are fields that can be merged (what is common for built-in functions and user functions) ⚠️ Anything touched here has to be double-tripled tested so as not to have any negative impact on overall performance.~~ (The way it is after this PR, with the info in one place and just the essential bits included in Function values, it's more than fine...)
- [x] In many cases, even though it may not make such a huge difference in the end, we are not using the best type possible. 
**For example:**
   - [x] Why is `.arity` an *int*? Could a function have 3 million parameters? (As a matter of fact, built-in functions are capped to 3 by definition)
   - [x] `.params` is a Value, meant to contain a Block, that contains a seq of Values, that are actually String objects, that contain a *string* (I guess if I think hard enough I could come up with a more complicated way of storing a simple array of strings... 😛 ) + we won't actually need the Value objects per-se; and if we did, we can always convert our strings to Value's internally. Ah, and on top of it, a very commonly used field, so the less the redirections we need, the better.
   - [x] ~~`.imports`, `.exports` fall under the same category - only we'd have to make sure the *nil*-ability tests stay in place (so, a pure, straight `seq` of strings is out of the question)~~ (The way it is right now is efficient enough, plus we maintain it's nil-ability tests as-is)
   - [x] ~~`.memoize` & `.inline` are definitely solely for user functions, but *perhaps* - given that there may be more bool-type of fields in the future - it could make sense to convert them to flags, as with the main Value objects~~ (Not very relevant in this case; even if we do want to add more `bool`s in there, we wouldn't gain anything by converting it to a set flags)

That being said, off to work! 😄 

## Type of change

- [x] Code cleanup

### CODE REVIEW

**Comment by beef331** [comment]
If all functions need those fields, there might not be much reason to move to ref.

Could `import` and `exports` be `seq[Option[string]]`? Given I've never wrote a line of Arturo I do not know what those relate to.

**Comment by drkameleon** [comment]
> If all functions need those fields, there might not be much reason to move to ref.
It depends: **all** built-in functions have `.args`, `.attrs`, `.returns` & `.example` (it is exactly what is shown when looking up the "info" of a specific command), but only the user functions that declare it somehow (in-code documentation) have this. All in all, their sole use is documentation:

Screenshot 2022-11-15 at 10 19 33

> Could `import` and `exports` be `seq[Option[string]]`? Given I've never wrote a line of Arturo I do not know what those relate to.

`.import` & `.export` are nothing but *optional* arguments (when creating a `function`) and include the symbols that are either force-included in scope, or the ones that will be exported to the outer scope (since - that aside - Arturo's function concept is in many ways quite "pure"):

https://arturo-lang.io/documentation/library/converters/function/

In any case, it's something that in 95% of the cases is not set at all.

**Comment by beef331** [comment]
Ah if it is just holding query information then yes you're correct It'll likely be much better for memory and performance.

**Comment by drkameleon** [comment]
@beef331 The above screenshot, annotated, so that you can get an idea of what is what - regarding the `VFunction` object:

Screenshot 2022-11-15 at 10 19 33

File: src/vm/values/types.nim [hunk 1]
```diff
@@ -228,11 +236,12 @@
...
                 line*: int
     ValueObj = typeof(Value()[])
+    FuncObj = typeof(VFunction()[])
 
 # Benchmarking
-
 {.hints: on.} # Apparently we cannot disable just `Name` hints?
-{.hint: "'Value's inner type is currently '" & $sizeof(ValueObj) & "'.".}
+{.hint: "Value's inner type is currently " & $sizeof(ValueObj) & ".".}
+{.hint: "Function's inner type is currently " & $sizeof(FuncObj) & ".".}
```
[the remark this answers is not in the document]
Comment (drkameleon) [reply src/vm/values/types.nim:244]: 🎆

File: src/vm/values/types.nim [hunk 2]
```diff
@@ -228,11 +240,12 @@
...
```
[the remark this answers is not in the document]
Comment (drkameleon) [reply src/vm/values/types.nim:248]: 🚀

**Comment by drkameleon** [comment]
Ready to merge! 🚀

**Pull request merged**